assets/src/hooks/v2/use_interval_paging.tsx

```diff
@@ -0,0 +1,64 @@
+import { useState, useEffect, useRef, useCallback } from "react";
+
+interface UseIntervalPagingProps {
+  numPages: number; // Total number of pages to cycle through
+  cycleIntervalMs?: number; // In milliseconds, the interval for cycling pages (only used if advanceOnDataRefresh is false)
+  onFinish: () => void; // Callback when cycling completes
+}
+
+/**
+ * This hook acts as a way for components with multiple pages to advance
+ * through their pages on a set time interval.
+ */
+function useIntervalPaging({
+  numPages,
+  cycleIntervalMs,
+  onFinish,
+}: UseIntervalPagingProps) {
+  const [pageIndex, setPageIndex] = useState(0);
+  // Use refs to keep stable references to numPages and interval without triggering re-renders
+  const intervalRef = useRef<NodeJS.Timeout | null>(null);
+  const numPagesRef = useRef(numPages);
+
+  useEffect(() => {
+    numPagesRef.current = numPages;
+  }, [numPages]);
+
+  // Callback that handles changing the state of pageIndex
+  const advancePage = useCallback(() => {
```

Comment:
nit: Since `advancePage` and `startTimer` are only used once, we could inline them and avoid the ceremony of `useCallback`.